### PR TITLE
Add new recipe to for Edgelock2GO auto registration

### PR DESCRIPTION
**Pull request by doanac**
This recipe requires a factory with Edgelock2GO enabled and
meta-subscriber-overrides additions for default.env and root.crt

Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
@rsalveti - this is what I've wound up with. It's your original recipe but I've changed the registration script to be in python, support aws-iot, and be a little better at being run more than once and recovering.

File: meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    except subprocess.CalledProcessError as e:
+        print("| " + e.output.decode().replace("\n", "\n| "), file=sys.stderr)
+        return False
+    log.info("EdgeLock2GO agent check-in complete")
+    return True
+
+
+class AwsKpHandler(NamedTuple):
+    key_id: str = "0X83000044"
+    crt_id: str = "0X83000045"
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register:142]: How do we know that the key/cert ids will stay the same?

File: meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        log.info("Processing keys for aws-iot...")
+        pkcs11.gen_kp("04", key_label)
+        with NamedTemporaryFile() as f:
+            SssCli.get_cert(self.crt_id, f.name)
+            pkcs11.write_object("cert", f.name, "05", crt_label)
+
+
+class AkliteKpHandler(NamedTuple):
+    key_id: str = "0X83000042"
+    crt_id: str = "0X83000043"
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register:159]: Same here.

**Comment by ricardosalveti** [comment]
Really nice, some minor comments but looking good. Mind just changing the commit subject prepending "base: " to it? Since meta-lmp has both the base and bsp layers, it is good to be explicit about it to make it easier to read the logs later.

File: meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register (hunk 1 above)
Comment (doanac) [reply meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register:142]: These are now managed via our API with constants:

 https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/blob/master/ota_lite/el2g.py#L13-16

It is a good idea for me to include those constants in our API docs. I'll put a change up for that with my team.

File: meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register.bb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+RDEPENDS:${PN} += "plug-and-trust-seteec python3-core python3-plug-and-trust-ssscli opensc"
+
+SRC_URI = " \
+	file://lmp-el2go-auto-register.service \
+	file://lmp-el2go-auto-register \
+	file://default.env \
+	file://root.crt \
```
Comment (quaresmajose) [inline meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register.bb:11]: These two are empty so we can achieve the same in `do_install` using the `touch` command and keep meta-lmp more clean

**Review by quaresmajose — APPROVED** [review]
LGTM with one minor point

File: meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register.bb (hunk 3 above)
Comment (doanac) [reply meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register.bb:11]: I stole from lmp-device-auto-register, so Ricardo can probably explain better. But,this approach is designed to let people override the file content in meta-subscriber-overrides. This recipe on its own is useless, the two empty files must be overridden with correct content. I don't think the "touch" approach would do that?

Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register.bb:11]: Yeah, the idea to have an empty file by default is that we can replace it by simply adding a file that has proper content under meta-subscriber (without having to add logic in a bbappend, besides FILESEXTRAPATHS).

File: meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register:142]: Cool, sounds good.

**Review by MrCry0 — APPROVED** [review]
LGTM

**Review by Tim-Anderson — APPROVED** [review]
LGTM

**Pull request merged**